### PR TITLE
fix: only send `rate_limit_per_user` parameter on thread creation if required

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

#746 2: electric boogaloo

It appears that text channels now support `default_thread_rate_limit_per_user` like forum channels (which, funnily enough, is documented in the api docs, since it was incorrectly documented initially and never removed).

This API change unfortunately breaks `channel.create_thread`/`message.create_thread` without `manage_threads` permissions, when the parent channel's `default_thread_rate_limit_per_user` is not `0`. Instead of sending `rate_limit_per_user: 0` by default, we want to inherit the parent's value now.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/channel.py [hunk 1]
```diff
@@ -871,10 +871,10 @@ async def create_thread(
 
             .. versionadded:: 2.3
 
-        slowmode_delay: :class:`int`
+        slowmode_delay: Optional[:class:`int`]
             Specifies the slowmode rate limit for users in this thread, in seconds.
             A value of ``0`` disables slowmode. The maximum value possible is ``21600``.
-            If not provided, slowmode is disabled.
+            If not provided or ``None``, slowmode is inherited from the parent channel.
```
Comment (onerandomusername) [inline disnake/channel.py:877]: the or none part here is confusing IMO

**Review by onerandomusername — CHANGES_REQUESTED** [review]
This doesn't actually solve the issue entirely. We're still sending `rate_limit_per_user` to the API with a null value even when the user does not provide any value.

File: disnake/channel.py (hunk 1 above)
Comment (shiftinv) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/818/commits/66d6206b059d29ce1f4cf1de09ed1ee2c6814dcc

**Comment by shiftinv** [comment]
> This doesn't actually solve the issue. We're still sending `rate_limit_per_user` to the API with a null value even when the user does not provide any value.

It does solve the issue, sending `rate_limit_per_user: null` works identically to not sending the field at all.
The different behaviors can be tested by setting the parent channel's `default_thread_rate_limit_per_user` to some non-zero value and trying to create threads without `manage_threads` perms.

It's documented as being nullable, but the actual semantics aren't documented (`?integer | amount of seconds a user has to wait before sending another message (0-21600)`), so I've updated that in https://github.com/DisnakeDev/disnake/pull/818/commits/9b02444e4d7f1d4bb155ab53c5b9d48d26730580 to be safe. It's effectively the same thing in the end, afaict.

**Comment by onerandomusername** [comment]
> It does solve the issue, sending `rate_limit_per_user: null` works identically to not sending the field at all. The different behaviors can be tested by setting the parent channel's `default_thread_rate_limit_per_user` to some non-zero value and trying to create threads without `manage_threads` perms.

I think that if we're fixing it, we should make it behave the same as creating a thread in forum channels: not sending the parameter if it is null.

**Comment by onerandomusername** [comment]
Ah, I didn't see your last sentence, nor look at https://github.com/DisnakeDev/disnake/commit/9b02444e4d7f1d4bb155ab53c5b9d48d26730580 as of writing that comment. I see now that we no longer send the parameter when its not set, which is good.

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**